src/main/kotlin/pl/edu/uj/ii/ksi/mordor/controllers/session/ReviewViewsFactory.kt

```diff
@@ -0,0 +1,137 @@
+package pl.edu.uj.ii.ksi.mordor.controllers.session
+
+import org.apache.commons.io.FileUtils
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.stereotype.Service
+import org.springframework.web.servlet.ModelAndView
+import org.springframework.web.servlet.view.RedirectView
+import org.springframework.web.util.UriUtils
+import pl.edu.uj.ii.ksi.mordor.controllers.FilesystemController
+import pl.edu.uj.ii.ksi.mordor.model.FileType as FileType
+import pl.edu.uj.ii.ksi.mordor.model.RelativeDirectory as RelativeDirectory
+import pl.edu.uj.ii.ksi.mordor.persistence.entities.Permission
+import pl.edu.uj.ii.ksi.mordor.services.IconNameProvider
+import pl.edu.uj.ii.ksi.mordor.services.repository.RepositoryDirectory
+import pl.edu.uj.ii.ksi.mordor.services.repository.RepositoryEntity
+import pl.edu.uj.ii.ksi.mordor.services.repository.RepositoryFile
+import pl.edu.uj.ii.ksi.mordor.services.upload.session.FileUploadSessionRepository
+
+@Service
+class ReviewViewsFactory(
```

Comment:
consider refactoring all preview* methods to avoid code duplication.

Reply:
Done